### PR TITLE
refactor/removed-player-input-as-far-as-possible

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description

**Fixes**: # (issue)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

**Test Configuration**:

### Checklist

- [ ] My code follows the [Coding Conventions](#coding-conventions)
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: Editor/CourseController/UI/CourseControllerSetupEditor.cs [hunk 1]
```diff
@@ -51,6 +52,13 @@ private void OnEnable()
             }
             
             currentRequiredComponents = availableCourseControllers[selectedIndex].GetRequiredSetupComponents();
+            currentRequiredComponents.AddRange(currentRequiredComponents
+                .SelectMany(type => type.GetCustomAttributes(typeof(RequireComponent)).Cast<RequireComponent>())
+                .SelectMany(component => new List<Type>() {component.m_Type0, component.m_Type1, component.m_Type2})
+                .Where(type => type != null)
+                .Distinct()
+                .Except(currentRequiredComponents)
+                .ToList());
```
Comment (Gusinuhe) [inline Editor/CourseController/UI/CourseControllerSetupEditor.cs:61]: Wow, I almost had a stroke trying to read that 🤕
What does this do? What does it for? What is the meaning of life?

Comment (SimonTheSourcerer) [reply Editor/CourseController/UI/CourseControllerSetupEditor.cs:61]: damn I failed

Comment (SimonTheSourcerer) [reply Editor/CourseController/UI/CourseControllerSetupEditor.cs:61]: it also adds all components required via RequiredComponent() to the currentRequiredComponents list.

Comment (Gusinuhe) [reply Editor/CourseController/UI/CourseControllerSetupEditor.cs:61]: image

Comment (SimonTheSourcerer) [reply Editor/CourseController/UI/CourseControllerSetupEditor.cs:61]: I think you did the same somewhere in editor, but i did not find the method :D

Comment (Gusinuhe) [reply Editor/CourseController/UI/CourseControllerSetupEditor.cs:61]: Na it's fine 😅

**Review by Gusinuhe — APPROVED** [review]

**Comment by Gusinuhe** [comment]
Please also remove `using UnityEngine.InputSystem;` and `using Innoactive.Creator.Core.Configuration;` from `SpectatorOverlay`.

**Review by Gusinuhe — APPROVED** [review]

**Pull request merged**